### ISSUE TITLE
Login does not work whatsoever

### ISSUE THREAD

**Issue by LillyWho** [sbpp/sourcebans-pp#835]
What are the steps to reproduce this issue?
-------------------------------------------
1. Set up SB++ according to the guide
2. Try to login with your new admin account

What happens?
-------------
The login doesn't fail with an error message, but you don't get logged in either.

What were you expecting to happen?
----------------------------------
Logging in successfully.

Any logs, error output, etc.?
----------------------------

No logs generated by php despite them being enabled, and no visible errors seem to occur on the page.

Any other comments?
-------------------
Web server is Apache2 using XAMPP

What versions of software are you using?
----------------------------------------
**Operating System:**
Windows Server 2022
**SourceBans++ Version:**
[SourceBans++]1.7.0 | Git: 1145
**PHP Version:**
8.1.12
**MySQL Version:**
MariaDB 10.4.27
**Link to your project:**
?
**Link to a phpinfo() output:**
https://gist.github.com/LillyWho/91aba142de89e95bfb806755e41b42be

**Comment by SlepPoweredBy2** [comment]
Yeah it also happened to me all of a sudden.

**Comment by Hackmastr** [comment]
Hello @LillyWho @SlepPoweredBy2 

What branch are you using? 
Thanks.

**Comment by LillyWho** [comment]
> Hello @LillyWho @SlepPoweredBy2
> 
> What branch are you using? Thanks.

I thought the git revision number and the PHP version gave it away that it's the branch for PHP 8.1

**Comment by F1F88** [comment]
I'm also having this issue now, does anyone know how to fix it?

Operating System: Windows Server 2012
SourceBans++ Version: 1.8.0
PHP Version: 8.1.1
MySQL Version: 8.0.12

**Comment by Rushaway** [comment]
@yy857 Hello. Do you have any PHP/JS errors or warnings (client or server side) when you try to login?

**Comment by F1F88** [comment]
> @yy857 Hello. Do you have any PHP/JS errors or warnings (client or server side) when you try to login?

No. I set "display_errors=On" in the php.ini, but I didn't find any error message when logging in.
After clicking Login, you will be taken to the Dashboard page, but you are still not logged in. I tried Clear Cookies, Incognito mode. It's all the same result

**Comment by F1F88** [comment]
I think this should be caused by 'Cross Origin Resource Sharing'.

A simple solution is to ensure that the domain name configured when adding a website matches the address accessed by the user in the browser. (Perhaps it's better to solve cross domain problems, but I'm not familiar with front-end development)

Recurrence: 
I use the nginx + php8.2 + mysql5.7 environment provided by the phpenv software.
SourceBans++ Version: 1.8.0 | Git: 1294

When adding a website, if the domain name does not match the address actually accessed in the browser. (For example, when adding a website, the domain name configured is sourcebanspp170, but the user uses localhost: 8080 in the browser to access the website).

Then it will appear that there is no error prompt after clicking the login button, but after jumping to the home page, it is still in the state of not logging in.

![image](https://github.com/sbpp/sourcebans-pp/assets/66011476/3c8c4c88-a5e5-4a79-8d7d-5db63b0a9e19)



If you delete the redirect of PLogin in includes/sb-callback.php, you can also see the message "Attempting to set a cookie through the Set Cookie header was blocked because its domain attribute is invalid for the current host URL" (translated from the login page screenshot below)

![image](https://github.com/sbpp/sourcebans-pp/assets/66011476/dc158836-ff34-41f1-94bb-f4912244a92e)

**Comment by TheOnlyDarkDragon** [comment]
Same issue is happening to me. Have you found any thing can fix this?

**Comment by DNA-styx** [comment]
> Same issue is happening to me. Have you found any thing can fix this?

If you have done a fresh install using the latest version make sure you run the updater: http://yourwebsite/updater . I wasn't able to log in when I did a recent install.


![Image](https://github.com/user-attachments/assets/97389a7b-71b1-4d60-a247-6cb62deb9f87)
